### PR TITLE
OSDOCS-10209: ROSA "Approved Access" (was Lockbox)

### PR DESCRIPTION
**Pull request by fmcdonal**
Version(s):
4.16+

Issue:
https://issues.redhat.com/browse/OSDOCS-10209

Link to docs preview:

- https://77370--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_release_notes/rosa-release-notes.html
- https://77370--ocpdocs-pr.netlify.app/openshift-rosa/latest/support/approved-access.html

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (Ginxo) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:23]: "Access Request Details" modal is already whenever you come from an email notification link

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (Ginxo) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:25]: would be "modal" a more accurate term? or is it not intended to be so specific here :thinking: 

```suggestion
. Click *Approve* or *Deny* under *Decision* in the *Access Request Details* modal.
```

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (Ginxo) [inline modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:27]: would be "modal" a more accurate term? or is it not intended to be so specific here :thinking: 

```suggestion
. Click *Approve* or *Deny* under *Decision* in the *Access Request Details* modal.
```

**Review by Ginxo — COMMENTED** [review]
well done @fmcdonal !! please check my comments

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (fmcdonal) [reply modules/support-reviewing-an-access-request-from-an-email-notification.adoc:23]: removed step 2, thanks

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (fmcdonal) [reply modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:27]: changed thanks

File: rosa_release_notes/rosa-release-notes.adoc
Comment (apahim) [inline rosa_release_notes/rosa-release-notes.adoc:19]: ... In the unlikely event should **Red Hat SRE** need access to customer data ...

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (apahim) [inline modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:31]: The sentence `will require further support case engagement` seems unnecessary, as it's not really a requirement. I'd phrase this whole note with something like:

```
Denying an access request will require you to complete the Justification field. In this case, Red Hat SRE will not be able to directly act on the resources related to the incident. Customers can still use the Red Hat Support to help investigate and fix the issue.
```

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (robpblake) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:15]: @fmcdonal The role is `Cluster Owner` that is required to approve/deny Access Requests.

File: support/approved-access.adoc
Comment (robpblake) [inline support/approved-access.adoc:12]:
```
Red{nbsp}Hat Site Reliability Engineering (SRE) managing and proactively supporting ROSA Clusters will typically not require access to systems containing customer data as part of normal operations. In the unlikely event SRE need access to systems containing customer data, the *Approved Access* functionality provides an interface for customers to review and *approve* or *deny* access to these systems.```

@fmcdonal Minor suggestion on wording here.

File: support/approved-access.adoc
Comment (robpblake) [inline support/approved-access.adoc:15]: @fmcdonal I think this section would benefit from a couple of additional points:

- We should inform customers that Approved Access is an opt-in service
- I think a section on "Why choose to Enable Approved Access?" would be useful to help customers make a decision e.g
   -  You have regulatory requirements that mean you must audit all access to systems potentially containing customer data.
   - ...
- Does it make more sense to inform the customer here about what happens if they do not respond to Access Requests?
  - Inform the customer that turning on Approved Access means that they need to respond in a timely manner otherwise there is a risk to the SLA of ROSA
  - Inform the customer what will happen if they deny Access to Red Hat SRE

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (robpblake) [inline modules/support-submitting-a-case-enable-approved-access.adoc:25]: @fmcdonal Enable `ROSA Approved Access`

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (robpblake) [inline modules/support-submitting-a-case-enable-approved-access.adoc:33]: @fmcdonal `Approved Access`

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (robpblake) [inline modules/support-submitting-a-case-enable-approved-access.adoc:36]: @fmcdonal  `Approved Access`

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (robpblake) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:30]: @Ginxo Can you confirm the wording here. Is it `Save` on the UI to submit a Decision for an Access Request?

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (robpblake) [inline modules/support-submitting-a-case-enable-approved-access.adoc:43]: @fmcdonal As a suggestion:


> With Approved Access enabled, for any Access Request created by Red{nbsp}Hat SRE, cluster owners will receive an email notification. The email notification will contain a link allowing the Cluster Owner to quickly approve or deny the Access Request.
> 
> Pending access requests are also available for review on the {hybrid-console-second}, within the cluster details page, via the Access Requests tab.
>

File: support/approved-access.adoc
Comment (robpblake) [inline support/approved-access.adoc:18]: @fmcdonal Should we add a section on "How am I informed of Access Requests?"

In that section we can explain:

- By default, the Cluster Owner will receive an email informing them of a new Access Request
  - If customers require additional users over/above the Cluster Owner to receive the email, they can add additional Cluster contacts using these steps: https://docs.openshift.com/rosa/observability/logging/sd-accessing-the-service-logs.html#adding-cluster-notification-contacts_sd-accessing-the-service-logs
- Pending Access Requests will be highlighted in the HCC via the clusters list or Access Requests tab on the cluster overview for the specific cluster

File: support/approved-access.adoc
Comment (fmcdonal) [reply support/approved-access.adoc:15]: @robpblake Do you feel this is not covered already in the "https://77370--ocpdocs-pr.netlify.app/openshift-rosa/latest/support/approved-access.html#support-reviewing-an-access-request-from-an-email-notification_approved-access" where we have notes about the "Deny" and "SLA"

Comment (robpblake) [reply support/approved-access.adoc:15]: @fmcdonal Yes it is covered in that section. My suggestion is we move it to the introductory section about Approved Access to allow customers to make an informed decision about whether or not to turn Approved Access on.

However customers approve/deny Access Requests, they should be made aware of the implications of being slow to respond as part of their consideration to turn it on.

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (fmcdonal) [reply modules/support-reviewing-an-access-request-from-an-email-notification.adoc:30]: Confirmed with Enrique it is `Save`

File: support/approved-access.adoc
Comment (fmcdonal) [reply support/approved-access.adoc:15]: applied

File: support/approved-access.adoc
Comment (fmcdonal) [reply support/approved-access.adoc:27]: applied

**Review by apahim — APPROVED** [review]
lgtm

**Review by Ginxo — APPROVED** [review]

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (jloss-redhat) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:23]: *Deny* requires a justification which I see is already noted but not in these steps.

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (bergerhoffer) [inline modules/support-submitting-a-case-enable-approved-access.adoc:39]: Just checking that "severity" really is lowercase in the UI, or if it should be capitalized?

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (bergerhoffer) [inline modules/support-submitting-a-case-enable-approved-access.adoc:8]: Just a suggestion to put the most important part (enabling approved access) first

```suggestion
= Enabling approved access for ROSA clusters by submitting a support case
```

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (bergerhoffer) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:23]: I don't know if we use "modal". Maybe "dialog"?

```suggestion
. In the *Access Request Details* dialog, click *Approve* or *Deny* under *Decision*.
```

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (bergerhoffer) [inline modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:10]: This should be inside the below ifdef as well. It doesn't seem to render the heading if there's nothing below it, but it shouldn't be stranded here just in case.

Also, I don't see this file being included anywhere that isn't ROSA - is there a reason for the below conditional, if it's not actually being used currently?

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (bergerhoffer) [inline modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:21]: Just checking if "Name" really needs to be bolded here, or if it's just choosing a cluster from the list, and could maybe just be normal "name" text?

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (bergerhoffer) [inline modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:27]: Same suggestions as the other file

```suggestion
. In the *Access Request Details* dialog, click *Approve* or *Deny* under *Decision*.
```

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (bergerhoffer) [inline modules/support-reviewing-an-access-request-from-an-email-notification.adoc:14]: Same comment about this conditional. Right now it's showing only if the distro is NOT ROSA or dedicated. Not sure if that's what was intended or not, but wondering if it should even be here if it's not being included in anything other than ROSA.

And also to put the prereqs heading inside the conditional to be safe

File: rosa_release_notes/rosa-release-notes.adoc
Comment (bergerhoffer) [inline rosa_release_notes/rosa-release-notes.adoc:19]: I forgot to submit this comment in the earlier review, but same comments I made in the other files could be applied to these as well.

This is also a pretty large block of text for users to read, so I'd recommend trying to break it up some. Even putting a "+" between these two paragraphs to make them separate paragraphs could help

**Review by bergerhoffer — COMMENTED** [review]
Forgot one comment.

Also just a reminder to squash commits before sending to merge review. And there is also a conflict that you'll need to resolve. Thanks!

File: modules/support-reviewing-an-access-request-from-an-email-notification.adoc
Comment (fmcdonal) [reply modules/support-reviewing-an-access-request-from-an-email-notification.adoc:23]: added it in here too:
"[NOTE]
====
Denying an access request will require you to complete the *Justification* field. In this case, SRE will not be able to directly act on the resources related to the incident. Customers can still use the link:https://access.redhat.com/support/cases/#/case/list[*Customer Support*] to help investigate and resolve any issues.
===="

File: support/approved-access.adoc
Comment (fmcdonal) [reply support/approved-access.adoc:3]: @robpblake or @arendej pls advise

File: support/approved-access.adoc
Comment (fmcdonal) [reply support/approved-access.adoc:12]: @robpblake are you happy with this new wording as I think you provided me with the content, thanks

File: support/approved-access.adoc
Comment (fmcdonal) [reply support/approved-access.adoc:16]: @robpblake which of the above 2 would you prefer please?

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (fmcdonal) [reply modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:9]: @arendej or @apahim can you please provide intros for both procedures as they need an intro before steps:

- https://77370--ocpdocs-pr.netlify.app/openshift-rosa/latest/support/approved-access#support-submitting-a-case-enable-approved-access_approved-access

- https://77370--ocpdocs-pr.netlify.app/openshift-rosa/latest/support/approved-access#support-reviewing-an-access-request-from-the-hybrid-cloud-console_approved-access

- https://77370--ocpdocs-pr.netlify.app/openshift-rosa/latest/support/approved-access#support-reviewing-an-access-request-from-an-email-notification_approved-access

, thanks

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (fmcdonal) [reply modules/support-submitting-a-case-enable-approved-access.adoc:39]: it was thanks

File: support/approved-access.adoc
Comment (robpblake) [reply support/approved-access.adoc:3]: Approved Access is the name of the feature. Consensus from today is to stay with that.

File: support/approved-access.adoc
Comment (robpblake) [reply support/approved-access.adoc:12]: @fmcdonal Small typo `normal operators` should be `normal operations`

File: support/approved-access.adoc
Comment (robpblake) [reply support/approved-access.adoc:16]: @fmcdonal I think the suggestion from @bergerhoffer is the easiest to read from my perspective 👍

File: _topic_maps/_topic_map_rosa.yml
Comment (fmcdonal) [reply _topic_maps/_topic_map_rosa.yml:312]: Spoke to team and want it to remain as is, thanks

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (fmcdonal) [reply modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:9]: @robpblake forgot to tag you here..

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (robpblake) [reply modules/support-submitting-a-case-enable-approved-access.adoc:9]: @fmcdonal :

> ROSA Approved Access is not enabled by default. To enable Approved Access for your ROSA clusters, you should create a support ticket.

File: modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc
Comment (robpblake) [reply modules/support-reviewing-an-access-request-from-the-hybrid-console.adoc:9]: @fmcdonal 

> Access Requests for your ROSA Clusters are reviewed via the Hybrid Cloud Console.

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (adellape) [inline modules/support-submitting-a-case-enable-approved-access.adoc:1]: Per https://github.com/openshift/openshift-docs/pull/77370/files#r1656901282 it seemed like the decision was to keep it init cap as "Approved Access" throughout, but this file mostly has it lowercase "approved access".

File: support/approved-access.adoc
Comment (adellape) [inline support/approved-access.adoc:16]: Here as well:
```suggestion
When Approved Access is enabled and an SRE creates an access request, _cluster owners_ receive an email notification informing them of a new access request. The email notification contains a link allowing the cluster owner to quickly approve or deny the access request. You must respond in a timely manner otherwise there is a risk to your SLA for {product-rosa}.
```

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (fmcdonal) [reply modules/support-submitting-a-case-enable-approved-access.adoc:1]: reverted as thought better too and what was agreed

File: modules/support-submitting-a-case-enable-approved-access.adoc
Comment (adellape) [inline modules/support-submitting-a-case-enable-approved-access.adoc:26]: `{hcp-title}` coming through verbatim

**Review by adellape — APPROVED** [review]

File: support/approved-access.adoc
Comment (adellape) [inline support/approved-access.adoc:12]: I'm not blocking merge on this, but overall per [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference), we can use underscores/italics for a "first reference" / "first term" style appearance of something (in this case, this first appearance of the feature name _Approved Access_), but after that it should just be Approved Access (no italics).

That is to say, I don't think considering it "general emphasis" is appropriate for subsequent instances.

image

However, if it ever shows up as an actual web console / UI element or page name or similar, then it should be asterisks/bold (which you have appropriately done elsewhere).

**Pull request merged**
